libraries/chain/block.cpp

```diff
@@ -111,4 +112,22 @@ namespace eosio { namespace chain {
       return validate_and_extract_block_extensions( block_extensions );
    }
 
+   signed_block_ptr pruned_block::to_signed_block() const {
+      if (prune_state != prune_state_type::complete_legacy)
+         return signed_block_ptr{};
+
+      auto result = std::make_shared<signed_block>(*this);
+      result->block_extensions = this->block_extensions;
+
+      auto visitor = overloaded{
+         [](const transaction_id_type &id) -> transaction_receipt::trx_type { return id; },
+         [](const pruned_transaction &trx) -> transaction_receipt::trx_type { 
+            return packed_transaction{trx.get_signed_transaction(), trx.get_compression()}; 
+      }};
+
+      for (const pruned_transaction_receipt &r : transactions){
+         result->transactions.emplace_back(transaction_receipt{r, r.trx.visit(visitor)});
+      }
```

Comment:
Can we add a `to_transaction_receipt` to `pruned_transaction_receipt` for this.

Reply:
.. and a `to_packed_transaction` to `pruned_transaction`

Reply:
`to_transaction_receipt` would have to return an `optional<transaction_receipt>` in that case (or throw an exception if called on a `pruned_transaction_receipt` that contained a `pruned_transaction` with prunable data that was not `full_legacy`). By only implementing `to_signed_block` as a public member function, we can handle the case where the `pruned_block` contains unconvertible data really cleanly: just return `signed_block_ptr{}`.